hpke/hpke.go

```diff
@@ -0,0 +1,173 @@
+// Package hpke implements hybrid public key encryption.
+package hpke
+
+import (
+	"crypto"
+)
+
+const versionLabel = "HPKE-06"
+
+// Exporter is
+type Exporter interface {
+	Export(expCtx []byte, len uint16) []byte
+}
+
+// Sealer is
+type Sealer interface {
+	Exporter
+	Seal(pt, aad []byte) (ct []byte, err error)
+}
+
+// Opener is
+type Opener interface {
+	Exporter
+	Open(ct, aad []byte) (pt []byte, err error)
+}
+
+// ModeID is
+type ModeID = uint8
+
+const (
+	Base    ModeID = iota // Base is
+	PSK                   // PSK is
+	Auth                  // Auth is
+	AuthPSK               // AuthPSK is
+)
+
+type Suite struct {
+	KemID  KemID
+	KdfID  KdfID
+	AeadID AeadID
+}
+
+type state struct {
+	Suite
+	modeID ModeID
+	skS    crypto.PrivateKey
+	pkS    crypto.PublicKey
+	psk    []byte
+	pskID  []byte
+}
+
+type Sender struct {
+	state
+	pkR  crypto.PublicKey
+	info []byte
+	seed []byte
+}
+
+func (s Suite) NewSender(pkR crypto.PublicKey, info, seed []byte) *Sender {
```

Comment:
Why use `crypto.PublicKey` instead of `kem.PublicKey"

Reply:
We plan to use crypto keys (from `"crypto"`) in the kem too.